src/front/wgsl/parse/mod.rs

```diff
@@ -0,0 +1,2066 @@
+use crate::front::wgsl::error::{Error, ExpectedToken, NumberType};
+use crate::front::wgsl::parse::lexer::{Lexer, Token};
+use crate::front::wgsl::parse::number::Number;
+use crate::front::SymbolTable;
+use crate::{Arena, FastHashSet, Handle, Span};
+
+pub mod ast;
+pub mod conv;
+pub mod lexer;
+pub mod number;
+
+/// State for constructing an AST expression.
```

Comment:
It might be worth putting in a comment:

```suggestion
/// State for constructing an AST expression.
///
/// Not to be confused with `lower::ExpressionContext`.
```